File: src/main/java/com/jesderr/shop/models/Buyer.java
package com.jesderr.shop.models;

import javax.persistence.*;

@Entity
@Table(name = "buyer")
public class Buyer {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    @Column(name = "name")
    public String name;
    @Column(name = "surname")
    public String surname;
    @Column(name = "gender")
    public String gender;
    @Column(name = "cardid")
    public Long bankCardId;

    public Buyer(Long id, String name, String surname, String gender, Long bankCardId) {
        this.id = id;
        this.name = name;
        this.surname = surname;
        this.gender = gender;
        this.bankCardId = bankCardId;
    }

    public Buyer() {

    }

}

File: shop-app/src/app/models/buyer.ts
export class Buyer {
  id: number | undefined;
  name: string | undefined;
  surname: string | undefined;
  gender: string | undefined;
  bankCardId: number | undefined;


  constructor(name: string, surname: string, gender: string) {
    this.name = name;
    this.surname = surname;
    this.gender = gender;
  }
}
